docs/developer-guide/etendo-classic/how-to-guides/how-to-create-build-validations-and-module-scripts.md

```diff
@@ -0,0 +1,349 @@
+
+###  Executing the module script
+
+The **module scripts** will be automatically executed in the `update.database` task, or in `update.database.mod` if the module is being applied. 
+
+!!!warning
+      Module scripts will not be compiled in this task, they need to have been compiled previously to be executed.
+
+Another important point to remark is that the module scripts work as **build validations** in regards to execution criteria; that is, they will be executed in every version of the module, or Core, in which they are present. The developer needs to take this into account (it is advisable not to design scripts which are **version-dependant**, they should always be designed to be **generic**).
+
+And, as it was explained above, scripts should also be designed to produce the same result if executed multiple times, because they will be executed every time the system is rebuilt, or the module is updated.
+
+!!!info
+      If you are adding a module script to Etendo Core, remember that you need to **include the binary classes** contained in `src-util/modulescript/build/classes/` to the repository as any source files are added. If not, they will not be executed (because the classes are not compiled by default in the build process, they are executed only if the binary files are there).
+
+###  Defining the module script execution limits
+
+The module scripts provide by default a couple of mechanisms that allow to control when they should be executed:
+
+####  Module version limits
+
+It is possible to define a dependency with two versions of a particular module to identify when the module script should be executed:
+
+  * **First Execution Version**: defines the first version since the module script should be executed. Before an upgrade, if the dependent module has a version lower or equal than this version, the module script will **NOT** be executed. By setting this limit we are ensuring that this version and previous ones do not require the module script execution. 
+
+  * **Last Execution Version**: defines the last version of the dependent module for which the module script should be executed. Before an upgrade, if the dependent module has a version higher or equal than this version, the module script will **NOT** be executed. 
+
+This way, the module script can be executed only when it is really needed, avoiding extra time calculations when updating database.
+
+![](../../../assets/developer-guide/etendo-classic/how-to-guides/How_to_create_build_validations_and_module_scripts-7.png)
+
+The dependent module and its limit versions can be set by overriding the`getModuleScriptExecutionLimits()` method in our `ModuleScript` subclass:
+
+```java
+@Override
+protected ModuleScriptExecutionLimits getModuleScriptExecutionLimits() {
+  return new ModuleScriptExecutionLimits("0", new OpenbravoVersion(3,0,27029), 
```

Comment:
It is still "OpenbravoVersion"?

Reply:
I looked for "EtendoVersion" or something related but couldn't find it.
Maybe @sebastianbarrozo or @valeg-etendo can confirm if there is anything else.

Reply:
To the best of my knowledge, this is still the case